Can't resolve 'react' in  document.js when Compatibility with React

Module not found: Error: Can't resolve 'react' in '/Users/yuanyan/Rax/rax-demo3/src/document'

```
info WEBPACK Starting the development server at:
info     http://localhost:9999/

info WEBPACK Hash: 3213a9b0c471faa8e8ef
info WEBPACK Version: webpack 4.41.5
info WEBPACK Child
info WEBPACK     Hash: 3213a9b0c471faa8e8ef
info WEBPACK     Time: 14662ms
info WEBPACK     Built at: 2020-01-06 21:12:13
info WEBPACK                       Asset      Size            Chunks  Chunk Names
info WEBPACK                _document.js  17.3 KiB
info WEBPACK                web/index.js  2.32 MiB             index  index
info WEBPACK     web/pages_Home_index.js  43.7 KiB  pages_Home_index  pages_Home_index
info WEBPACK     Child
info WEBPACK                Asset      Size    Chunks  Chunk Names
info WEBPACK         _document.js  17.3 KiB  document  document
info WEBPACK
info WEBPACK         ERROR in ./src/document/index.jsx
info WEBPACK         Module not found: Error: Can't resolve 'react' in '/Users/yuanyan/Rax/rax-demo3/src/document'
info WEBPACK          @ ./src/document/index.jsx 1:0-38 7:5-18 9:5-18 9:32-45
info WEBPACK          @ multi ./src/document/index.jsx
info WEBPACK Compiled successfully
Set environment `DEBUG=true` to see detail error stacks.
html path is not found
```

```jsx
import { createElement } from 'react';
import { Root, Style, Script} from 'rax-document';

function Document() {
  return (
    <html>
      <head>
        <meta charset="utf-8" />
        <meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1,minimum-scale=1,user-scalable=no,viewport-fit=cover"/>
        <title>rax-demo3</title>
        <Style />
      </head>
      <body>
        {/* root container */}
        <Root />
        <Script />
      </body>
    </html>
  );
}
export default Document;
```

--- a/packages/build-plugin-rax-app/package.json
+++ b/packages/build-plugin-rax-app/package.json
@@ -1,6 +1,6 @@
 {
   "name": "build-plugin-rax-app",
-  "version": "1.0.4",
+  "version": "1.0.5",
   "description": "rax app base plugins",
   "main": "src/index.js",
   "scripts": {

--- a/packages/build-plugin-rax-app/src/plugins/UniversalDocumentPlugin.js
+++ b/packages/build-plugin-rax-app/src/plugins/UniversalDocumentPlugin.js
@@ -45,7 +45,11 @@ module.exports = class UniversalDocumentPlugin {
     // web/[name].js => web
     const targetOutputDir = path.dirname(outputFileName);
 
-    const documentWebpackConfig = getWebpackConfigForDocument(this.context, absoluteDocumentPath, config.output.path);
+    const documentWebpackConfig = getWebpackConfigForDocument(this.context, {
+      entry: absoluteDocumentPath,
+      outputPath: config.output.path,
+      alias: config.resolve.alias, // sync the alias, eg. react, react-dom
+    });
 
     let fileDependencies = [];
     let documentContent;
@@ -119,22 +123,27 @@ module.exports = class UniversalDocumentPlugin {
 
 /**
  * custom webpack config for document
- * @param {*} documentPath  document source path
- * @param {*} dest dest path
+ * @param {*} context build plugin context
+ * @param {*} options options for webpack config
  */
-function getWebpackConfigForDocument(context, documentPath, dest) {
+function getWebpackConfigForDocument(context, options) {
+  const { entry, outputPath, alias } = options;
   const webpackChainConfig = getSSRBaseConfig(context);
 
   webpackChainConfig
     .entry('document')
-    .add(documentPath);
+    .add(entry);
 
   webpackChainConfig.output
-    .path(dest)
+    .path(outputPath)
     .filename(TEMP_FLIE_NAME);
 
   webpackChainConfig.externals({
     rax: 'rax',
+  });
+
+  Object.keys(alias).forEach((key) => {
+    webpackChainConfig.resolve.alias.set(key, alias[key]);
   });
 
   const documentWebpackConfig = webpackChainConfig.toConfig();

--- a/packages/build-plugin-rax-app/src/plugins/UniversalDocumentPlugin.js
+++ b/packages/build-plugin-rax-app/src/plugins/UniversalDocumentPlugin.js
@@ -48,7 +48,7 @@ module.exports = class UniversalDocumentPlugin {
     const documentWebpackConfig = getWebpackConfigForDocument(this.context, {
       entry: absoluteDocumentPath,
       outputPath: config.output.path,
-      alias: config.resolve.alias, // sync the alias, eg. react, react-dom
+      alias: config.resolve ? config.resolve.alias : {}, // sync the alias, eg. react, react-dom
     });
 
     let fileDependencies = [];
@@ -124,7 +124,7 @@ module.exports = class UniversalDocumentPlugin {
 /**
  * custom webpack config for document
  * @param {*} context build plugin context
- * @param {*} options options for webpack config
+ * @param {*} options defalut config for webpack
  */
 function getWebpackConfigForDocument(context, options) {
   const { entry, outputPath, alias } = options;

